src/file.rs

```diff
@@ -93,6 +98,24 @@ impl File {
     }
 }
 
+/// Read the content and return as a [String] if it can be decoded as one of the supported encodings from [Encoding].
+pub fn read(input: impl Read) -> Result<String, FileError> {
+    read_to_text_data(input).map(|content| content.data)
```

Comment:
You could `map` or you could `?` and I think the end result is the same.
```suggestion
    read_to_text_data(input)?.data
```